### PR TITLE
Implement #369 feature (reportes in HTTP extension).

### PR DESCRIPTION
**Pull request by la10736**
Requester don't need to use busy list. Implemented waiting dictionary and block request
state reset on reset_all requests and polling http errors events.
1. busy information was already in waiting dictionary
2. On reset_all or connection error (polliing error) is better reset all reporter blocks state

### CODE REVIEW

**Comment by sclements** [comment]
@cwillisf can you take a look at this? we need to make sure this doesn't break Javascript extensions.

File: src/extensions/ExtensionManager.as [hunk 1]
```diff
@@ -442,12 +445,9 @@ public class ExtensionManager {
...
 		function responseHandler(e:Event):void {
-			if(e.type == Event.COMPLETE)
-				b.response = loader.data;
-			else
-				b.response = '';
-
-			b.requestState = 2;
+			if (e.type != Event.COMPLETE){
+				b.requestState = 2;
+			}
```
Comment (cwillisf) [inline src/extensions/ExtensionManager.as:450]: Shouldn't this still collect `loader.data`?

File: src/extensions/ExtensionManager.as [hunk 2]
```diff
@@ -162,15 +170,11 @@ public class ExtensionManager {
 		var ext:ScratchExtension = extensionDict[extensionName];
 		if (ext == null) return; // unknown extension
 
-		var index:int = ext.busy.indexOf(id);
```
Comment (cwillisf) [inline src/extensions/ExtensionManager.as:165]: I'm nervous about removing the use of `ext.busy` for reporters / requesters. It looks OK at first glance but if possible I'd prefer a more conservative change that only implements HTTP requesters without affecting the way we track requesters in general.

File: src/extensions/ExtensionManager.as [hunk 3]
```diff
@@ -571,6 +572,13 @@ public class ExtensionManager {
 					if (ext.busy.indexOf(id) == -1) ext.busy.push(id);
 				}
 				break;
+			case '_result':
+				i = value.indexOf(' ');
+				if (i == -1) i = value.length;
+				var resid:int = parseInt(value.slice(0, i));
+				var retval:String = value.slice(i + 1);
```
Comment (cwillisf) [inline src/extensions/ExtensionManager.as:579]: It may be possible to simplify this by using `value.split` as in the `'_busy'` case.

**Review by cwillisf — CHANGES_REQUESTED** [review]
Sorry this took so long to get to. We're postponing many improvements until Scratch 3.0, but I think this one might make sense to integrate into 2.0 for those still using HTTP extensions.

**Comment by la10736** [comment]
I'll try to take a look to it in the next days... after 1.5 years without working to it is hard to recover all details.

**Comment by thisandagain** [comment]
Thanks for the contribution!

Scratch 2.0 is now in maintenance mode while the team focuses efforts on [Scratch 3.0](https://llk.github.io/scratch-gui). While critical issues will be addressed please note that any feature requests or minor issues will not be reviewed until 3.0 is released. Because of this, we will be closing this PR and only accepting patches to `scratch-flash` that resolve bugs labeled as `prio-critical`. 

If you are interested in contributing to Scratch 3.0 please take a look at the following repositories:
https://github.com/llk/scratch-gui
https://github.com/llk/scratch-blocks
https://github.com/llk/scratch-vm
https://github.com/llk/scratch-render
https://github.com/llk/scratch-audio

**Pull request closed without merging**